src/CustomComponents/CustomInput.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { CSSProperties , ReactEventHandler, useState } from 'react'
+
+interface CustomInputProps
+{
+    style?: React.CSSProperties;
+    hoverStyle?: React.CSSProperties;
+    borderRadius: string;
+    placeholder: string;
+    type: string;
+    id?: string;
+    onChange?: ReactEventHandler;
+}
+
+
+const CustomInput : React.FC<CustomInputProps> = ({style, placeholder , borderRadius, hoverStyle, id , onChange , type}) =>
+{
+    const [isInputHovered , setIsInputHovered] = useState(false);
```

Comment:
why have used useState here ? it is good but why ?

Reply:
Using useState in the CustomInput component allows you to manage and update the state of isInputHovered, which determines whether the input element is currently being hovered over by the mouse (true) or not (false), IN addition, we have that this isInputHovered is for the single component CustomInput so we are not required to make it in a store , because this state is not shared across multiple components